### PR TITLE
Feat/sortable grid list component

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Based on https://github.com/wix/react-native-ui-lib/pull/1914

Implement a SortableGridList component based on GridList logic
This component is using ScrollView and not a FlatList due to gesture constraints

## Changelog
New SortableGridList component!

### CODE REVIEW

File: demo/src/screens/componentScreens/SortableGridListScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  Card,
+  Spacings,
+  BorderRadiuses,
+  GridListProps,
+  SortableGridListProps
+} from 'react-native-ui-lib';
+
+import products from '../../data/products';
+
+class GridListScreen extends Component {
```
Comment (lidord-wix) [inline demo/src/screens/componentScreens/SortableGridListScreen.tsx:17]: Should be SortableGridListScreen

File: src/components/sortableGridList/usePresenter.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export const animationConfig = {
+  easing: Easing.inOut(Easing.ease),
+  duration: 350
+};
+
+export type ItemsLayouts = ({width: number; height: number} | undefined)[];
+export type ItemsOrder = number[];
+
+const usePresenter = (itemsSize: number, numOfColumns: number, itemSpacing: number) => {
```
Comment (lidord-wix) [inline src/components/sortableGridList/usePresenter.ts:17]: By `itemsSize` do you mean the number of items?

File: src/components/sortableGridList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    onOrderChange?.(newData, itemsOrder.value);
+  }, [onOrderChange, data]);
+
+  const _renderItem = useCallback(({item, index}) => {
+    const lastItemInRow = (index + 1) % numberOfColumns === 0;
+    return (
+      <SortableItem
+        key={index}
+        {...presenter}
+        style={[itemContainerStyle, lastItemInRow && {marginRight: 0}]}
```
Comment (lidord-wix) [inline src/components/sortableGridList/index.tsx:45]: You can use `useMemo` here

**Review by lidord-wix — COMMENTED** [review]
I left a few small comments, and:
- please add an API file.
- worth mentioning somewhere that it supports only square items.
- don't forget to make a private PR for exporting the new components and adding the screens once this one is ready.

File: src/components/sortableGridList/usePresenter.ts (hunk 2 above)
Comment (ethanshar) [reply src/components/sortableGridList/usePresenter.ts:17]: true (: 
Think I should rename?

Comment (lidord-wix) [reply src/components/sortableGridList/usePresenter.ts:17]: Yup :)

File: src/components/sortableGridList/index.tsx (hunk 3 above)
Comment (ethanshar) [reply src/components/sortableGridList/index.tsx:45]: I can't really 😬  it depends on a value exists in the _renderItem scope (lastItemInRow is based on index argument)

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**